go/libraries/doltcore/table/typed/parquet/writer.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parquet
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/xitongsys/parquet-go-source/local"
+	"github.com/xitongsys/parquet-go/source"
+	"github.com/xitongsys/parquet-go/writer"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/row"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema/typeinfo"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/sqlutil"
+)
+
+var WriteBufSize = 256 * 1024
+
+type ParquetWriter struct {
+	filewriter source.ParquetFile
+	pwriter    *writer.CSVWriter
+	sch        schema.Schema
+}
+
+func NewParquetWriter(outSch schema.Schema, destName string) (*ParquetWriter, error) {
+	allCols := outSch.GetAllCols()
+	columns := allCols.GetColumns()
+
+	ColTypeMap := getTypeMap()
+	var csvSchema []string
+	var repetitionType string
+	for _, col := range columns {
+		colName := col.Name
+		colType := col.TypeInfo.GetTypeIdentifier()
+		if col.IsNullable() {
+			repetitionType = `, repetitiontype=OPTIONAL`
+		} else {
+			repetitionType = ``
+		}
+		csvSchema = append(csvSchema, fmt.Sprintf("name=%s, %s%s", colName, ColTypeMap[colType], repetitionType))
+	}
+
+	fw, err := local.NewLocalFileWriter(destName)
+	if err != nil {
+		return nil, err
+	}
+
+	// np configures the degree of concurrency for our Reader and Writers
+	// TODO: not sure what default value to set 'np' to
```

Comment:
Remove the TODO and add a comment that `4` was the recommended value of np from the package